deploy/marketplace.csv.yaml

```diff
@@ -0,0 +1,102 @@
+apiVersion: operators.coreos.com/v1alpha1
```

Comment:
Do we want people to deploy with the csv? It is possible to deploy by applying a couple of manifest files in the deploy folder, but it's meant to be used with OLM. Thoughts?

Reply:
One has to follow these steps to deploy with OLM 
- `kubectl apply the CRD file(s)`
- `kubectl apply the CSV file`

It might be worth adding this to a new section in README. Your call :)

Reply:
No, I think it'd be good to add. I'll add something like this:

### Deploying the Marketplace Operator with OLM

#### In an OKD Cluster
Start by making sure that the namespace in the `deploy/rbac.yaml` and `deploy/csv/marketplace.csv.yaml` are the same.
```bash
$ oc apply -f deploy/catalogsourceconfig.crd.yaml 
$ oc apply -f deploy/operatorsource.crd.yaml
$ oc apply -f deploy/rbac.yaml
$ oc apply -f deploy/csv/marketplace.csv.yaml
```
#### In an Kubernetes Cluster
...

Reply:
So I had made a similar assumption which delayed this CR. In 7.1, OLM would create the roles for the given service account if you ran `oc apply -f deploy/csv/marketplace.csv.yaml` but this changed in 7.2, which requires a Subscription or install. I tested this locally below:

```
$  oc new-project marketplace
...
$ oc apply -f catalogsourceconfig.crd.yaml 
customresourcedefinition.apiextensions.k8s.io/catalogsourceconfigs.marketplace.redhat.com created

$ oc apply -f operatorsource.crd.yaml 
customresourcedefinition.apiextensions.k8s.io/operatorsources.marketplace.redhat.com created

$ oc apply -f csv/marketplace.csv.yaml 
clusterserviceversion.operators.coreos.com/marketplace-operator.v0.0.1 created

$ oc get pods
No resources found.
$ oc describe csv marketplace-operator.v0.0.1 
Name:         marketplace-operator.v0.0.1
Namespace:    marketplace
Labels:       <none>
Annotations:  kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"operators.coreos.com/v1alpha1","customresourcedefinitions":{"owned":[{"description":"Represents an Operator.","displayName":"Operator So...
API Version:  operators.coreos.com/v1alpha1
Kind:         ClusterServiceVersion
Metadata:
...
Status:
  Conditions:
    Last Transition Time:  2018-11-16T18:55:23Z
    Last Update Time:      2018-11-16T18:55:23Z
    Message:               requirements not yet checked
    Phase:                 Pending
    Reason:                RequirementsUnknown
    Last Transition Time:  2018-11-16T18:55:23Z
    Last Update Time:      2018-11-16T18:55:23Z
    Message:               one or more requirements couldn't be found
    Phase:                 Pending
    Reason:                RequirementsNotMet
  Last Transition Time:    2018-11-16T18:55:23Z
  Last Update Time:        2018-11-16T18:55:23Z
  Message:                 one or more requirements couldn't be found
  Phase:                   Pending
  Reason:                  RequirementsNotMet
  Requirement Status:
    Dependents:
      Group:    rbac.authorization.k8s.io
      Kind:     PolicyRule
      Message:  cluster rule:{"verbs":["*"],"apiGroups":["marketplace.redhat.com"],"resources":["*"]}
      Status:   NotSatisfied
      Version:  v1beta1
      Group:    rbac.authorization.k8s.io
      Kind:     PolicyRule
      Message:  cluster rule:{"verbs":["*"],"apiGroups":[""],"resources":["services","configmaps"]}
      Status:   NotSatisfied
      Version:  v1beta1
      Group:    rbac.authorization.k8s.io
      Kind:     PolicyRule
      Message:  cluster rule:{"verbs":["*"],"apiGroups":["operators.coreos.com"],"resources":["catalogsources"]}
      Status:   NotSatisfied
      Version:  v1beta1
    Group:      
    Kind:       ServiceAccount
    Name:       default
    Status:     PresentNotSatisfied
    Version:    v1
Events:         <none>
```

Reply:
Makes sense, thank you for the explanation.

Reply:
No problem! Thanks for the review! Will update it in the next PR!

Reply:
@tkashem the most recent commit should have the changes you requested in. Could you resolve this conversation if it meets your requirements?

Reply:
looks good